src/fsharp/SyntaxTree.fs

```diff
@@ -507,6 +507,7 @@ type SynType =
     /// F# syntax: const expr, used in static parameters to type providers
     | StaticConstantExpr of
         expr: SynExpr *
+        exprRange: range *
```

Comment:
Is there a need for this field? Let's calculate this range from expr, like it's done in `Do`, `Lazy`, and `Assert`?